pandas/core/indexes/range.py

```diff
@@ -187,6 +187,15 @@ def _format_data(self, name=None):
         # we are formatting thru the attributes
         return None
 
+    def _format_with_header(self, header: List[str], na_rep: str = "NaN") -> List[str]:
+        if len(self._range) == 0:
+            return []
```

Comment:
should this be `return header`?

Reply:
Yes.

I added tests for this, and found that ``DateTimeIndex([]).format(name=True)`` returns ["None"], where it should return [""]. Same for PeriodIndex. I fixed that bug.